### ISSUE TITLE
COLLECTING DATA

### ISSUE THREAD

**Issue by Akai19** [thu4n/ESP32-WiFi-Sensing#21]
I have problems in collecting the data

1. Im using window and I use the command  'idf.py monitor | findstr "CSI_DATA" > my-experiment-file.csv' but there is no output so Itried 
2. idf.py monitor  | tee >trial13.csv  now the saved collected csi data is in one column 

[trial13.csv](https://github.com/user-attachments/files/17372373/trial13.csv)

**Comment by thu4n** [comment]
Hi, your issue is quite similar to https://github.com/StevenMHernandez/ESP32-CSI-Tool/issues/111 . Assuming you use the toolkit without any modification, I can't really help since I couldn't reproduce it on my side. I suggest you go the link and state your similar problem there.

**Comment by Akai19** [comment]
yes we are in the same group. if ever, I think the problem is how the collected data is saved in my laptop, will there be a problem in collecting data if I directly collect it in esp32? I kinda have no idea, if ever the problem is in my laptop then we can just manually delimited and put it into machine learning or???

**Comment by thu4n** [comment]
There won't be a problem with directly collecting CSI data in the esp32, you only need to make sure it has enough storage space and a way to get the data later. Delimiting the data manually is fine too, just make sure the CSI data column isn't changed by the process.

**Comment by Akai19** [comment]
An update:
- we have collected CSI DATA successfully. There are no problems with the coding at all. We used the wrong powershell that is why the findstr is not working. When we decided to switch and command the findstr in CMD, the collected CSI DATA was saved in correct CSV which is it is not in one column anymore, the data's are now delimited. We tried some CSI DATAs and tried filtering and it was also successful too; we are now moving in machine learning; we will update you more if we encounter an issue again. For now, we can close this issue.

_**The only problem is that the findstr doesn't work in Esp-IDF Powershell so we switch to CMD.**_

**Comment by Mohammed-Baqir** [comment]
![ISSUE1_SAVEDATA](https://github.com/user-attachments/assets/b5171bd4-7c6d-47b6-9112-fd7f9e2602ed)
any one got this kind of errors ?

PS E:\esp-v4.3\esp-idf\esp32-csi-tool\active_ap>  idf.py -p \\.\COM3 monitor | findstr "CSI_DATA" > my-experiment-file.c
sv
--- idf_monitor on \\.\COM3 921600 ---
--- Quit: Ctrl+] | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H ---
0x400813f8: call_start_cpu1 at E:/esp-v4.3/esp-idf/components/esp_system/port/cpu_start.c:141

**Comment by zxfrq** [comment]
Hello, I am using esp32 s3 series chip to collect data, under 40MHZ, my data length is 384, after getting the amplitude and phase, they have 192 columns respectively, but when I refer to your data, I found that the data on your side is unified into 200x64, my data contains 26 columns of data columns which are all 0, please ask me to directly remove these 0 columns, and use the same method on the data. Can I directly remove these 0 columns and use the filtering and noise reduction on the data, can I directly use it for model training, which I am very confused about?
